src/TimePicker.js

```diff
@@ -112,20 +112,32 @@ const __class = declare('argos.TimePicker', [_Widget, _Templated], {
     '55',
   ],
 
-  createHourModal: function createHourModal() {
-    this._hourModal = new Modal({ id: 'hour-modal ' + this.id, showBackdrop: false, positioning: 'right' });
-    this._hourModal.placeModal(this.domNode.offsetParent)
-                   .setContentPicklist({ items: this.hourValues, action: 'setSelectedHour', actionScope: this });
-    this.setSelectedHour({ target: this._hourModal.getContent().children[0] });
-    connect.connect(this.hourNode, 'onclick', this, this.toggleHours);
+  createHourModal: function createHourModal(initial) {
+    if (!this._hourModal) {
+      this._hourModal = new Modal({ id: 'hour-modal ' + this.id, showBackdrop: false, positioning: 'right' });
+      this._hourModal.placeModal(this.domNode.offsetParent)
+                     .setContentPicklist({ items: this.hourValues, action: 'setSelectedHour', actionScope: this });
+      connect.connect(this.hourNode, 'onclick', this, this.toggleHours);
+    }
+    this.setSelectedHour({ target: this._hourModal.getContent().children[this.hourValues.indexOf(initial)] });
     return this;
   },
-  createMinuteModal: function createMinuteModal() {
-    this._minuteModal = new Modal({ id: 'minute-modal ' + this.id, showBackdrop: false, positioning: 'right' });
-    this._minuteModal.placeModal(this.domNode.offsetParent)
-                     .setContentPicklist({ items: this.minuteValues, action: 'setSelectedMinute', actionScope: this });
-    this.setSelectedMinute({ target: this._minuteModal.getContent().children[0] });
-    connect.connect(this.minuteNode, 'onclick', this, this.toggleMinutes);
+  createMinuteModal: function createMinuteModal(initial) {
+    if (!this._minuteModal) {
+      this._minuteModal = new Modal({ id: 'minute-modal ' + this.id, showBackdrop: false, positioning: 'right' });
+      this._minuteModal.placeModal(this.domNode.offsetParent)
+                       .setContentPicklist({ items: this.minuteValues, action: 'setSelectedMinute', actionScope: this });
+      connect.connect(this.minuteNode, 'onclick', this, this.toggleMinutes);
+    }
+    if (initial % 5 === 0) {
```

Comment:
Should we even have a special case for values that are divisible by 5? The else block handles those as well anyways. It would remove the duplicate `this.setSelectedMinute` calls in each block. Also this block isn't checking for values over 60 like the else block.